Passing mutable variable to functions

**V version:** 
0.1.3

**What did you do?**
```
fn multiply_by_2(num mut int) {
    num = num * 2
}

fn main() {
    num := 1
    multiply_by_2(mut num)
    println(num)
}
```

**What did you expect to see?**
``` 2 ```
 
**What did you see instead?**
```
/home/Henrixounez//.vlang//test2.c: In function ‘multiply_by_2’:
/home/Henrixounez//.vlang//test2.c:3577:14: error: invalid operands to binary * (have ‘int *’ and ‘int’)
  num  =  num * 2 ;
              ^
V panic: clang error
```



This issue is a follow up of the problem discussed in comments of #549 and #581

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -1193,7 +1193,7 @@ fn (p mut Parser) name_expr() string {
 	hack_lit := p.lit
 	// amp
 	ptr := p.tok == AMP
-	deref := p.tok == MUL
+	mut deref := p.tok == MUL
 	if ptr || deref {
 		p.next()
 	}
@@ -1237,6 +1237,7 @@ fn (p mut Parser) name_expr() string {
 	// Variable
 	v := p.cur_fn.find_var(name)
 	if v.name.len != 0 {
+		deref = deref || (v.is_arg && v.is_mut && is_mutable_type(v.typ))
 		if ptr {
 			p.gen('& /*vvar*/ ')
 		}

--- a/compiler/scanner.v
+++ b/compiler/scanner.v
@@ -713,3 +713,24 @@ fn (s mut Scanner) create_type_string(T Type, name string) {
 fn (p mut Parser) create_type_string(T Type, name string) {
 	p.scanner.create_type_string(T, name)
 }
+
+//Checks if mut type can be dereferenced to be modified
+fn is_mutable_type(typ string) bool {
+	if typ == 'bool*' {
+		return true
+	}
+	if typ == 'int*' || typ == 'rune*' || typ == 'i8*' ||
+	   typ == 'i16*' || typ == 'i32*' || typ == 'i64*' {
+		return true
+	}
+	if typ == 'byte*' || typ == 'u8*' || typ == 'u16*' ||
+	   typ == 'u32*' || typ == 'u64*' {
+		return true
+	}
+	if typ == 'f32*' || typ == 'f64*' {
+		return true
+	}
+	if typ == 'string*' {
+		return true
+	}
+}

